cpp/log/frontend.cc

```diff
@@ -111,95 +105,40 @@ std::string Frontend::SubmitResultString(SubmitResult result) {
 }
 
 // static
-SubmitResult Frontend::GetSubmitError(
-    CertSubmissionHandler::SubmitResult result) {
+Status Frontend::GetSubmitError(CertSubmissionHandler::SubmitResult result) {
   CHECK_NE(result, CertSubmissionHandler::OK);
 
-  SubmitResult submit_result;
   switch (result) {
     case CertSubmissionHandler::EMPTY_SUBMISSION:
+      return util::Status(util::error::INVALID_ARGUMENT, "Empty submission.");
     case CertSubmissionHandler::INVALID_PEM_ENCODED_CHAIN:
-      submit_result = BAD_PEM_FORMAT;
-      break;
+      return util::Status(util::error::INVALID_ARGUMENT,
+                          "Invalid PEM encoded chain.");
     case CertSubmissionHandler::SUBMISSION_TOO_LONG:
-      submit_result = SUBMISSION_TOO_LONG;
-      break;
+      return util::Status(util::error::INVALID_ARGUMENT,
+                          "Submission too long.");
     case CertSubmissionHandler::INVALID_CERTIFICATE_CHAIN:
+      return util::Status(util::error::FAILED_PRECONDITION,
```

Comment:
That might be more ```INVALID_ARGUMENT```? Going by this:

https://github.com/google/certificate-transparency/blob/super_duper/cpp/util/status.h#L75